### PR TITLE
Remove extra parens wrapping for type applications that contain arrow

### LINKED ISSUE #1616: Printer: extra parentheses around function in parametric types
```reason
type foo = option(int => int);
```

Turns into

```reason
type foo = option((int => int));
```

### PR DESCRIPTION
**Pull request by chenglou**
Fixes #1616

### CODE REVIEW

**Comment by jordwalke** [comment]
This is probably okay, but can you copy/paste the test cases you added, and add attributes at various parts on the types (both inside and outside the parens etc)? The code you're calling into now deals with attributes and I want to make sure this didn't break any of those cases.

File: formatTest/unit_tests/expected_output/wrappingTest.re [hunk 1]
```diff
@@ -2514,11 +2514,10 @@ type wrappingGadt(_) =
...
                                    )
-  | Add: wrappingGadt(((int, int) => int))
-  | App(
-         wrappingGadt(('b => 'a)),
-         wrappingGadt('b)
-       ): wrappingGadt('a);
+  | Add: wrappingGadt((int, int) => int)
+  | App(wrappingGadt('b => 'a), wrappingGadt('b)): wrappingGadt(
+                                        'a
+                                        );
```
Comment (IwanKaramazow) [inline formatTest/unit_tests/expected_output/wrappingTest.re]: 🙈

Comment (chenglou) [reply formatTest/unit_tests/expected_output/wrappingTest.re]: Unrelated right

Comment (jordwalke) [reply formatTest/unit_tests/expected_output/wrappingTest.re]: Yeah what’s up there.

Comment (IwanKaramazow) [reply formatTest/unit_tests/expected_output/wrappingTest.re]: I can fix this later today

**Comment by chenglou** [comment]
I've put some attributes on everything

**Comment by IwanKaramazow** [comment]
Looks solid, should be ok.

**Pull request merged**